expo-zustand-styled-components/src/services/get-repo-file.ts

```diff
@@ -16,12 +16,19 @@ type Response = {
 };
 
 const getRepoFile = async (variables: RepoFileVariables) => {
-  try {
-    useAppStore.setState({ isLoading: true });
-    const resp = (await FetchApi({ query: REPO_FILE_QUERY, variables })) as Response;
-    useAppStore.setState({ isLoading: false, file: resp?.data?.repository?.blob });
-  } catch (err) {
-    useAppStore.setState({ isLoading: false, error: err.message });
+  const key = `${variables.name}-${variables.owner}-${variables.expression}`;
+  if (useRepoInfoStore.getState()._file.has(key)) {
+    const data = useRepoInfoStore.getState()._file.get(key);
+    useRepoInfoStore.setState({ isLoading: false, file: data });
+  } else {
+    try {
+      useRepoInfoStore.setState({ isLoading: true });
+      const resp = (await FetchApi({ query: REPO_FILE_QUERY, variables })) as Response;
```

Comment:
Just a suggestion and maybe for another PR. I feel it will make a lot of sense for FetchAPI to take a take generic type of the data it is meant to return. 
So an implementation like 
```ts
export function FetchApi<T>(): T {
// ....
}
```
And then it is easier to just pass the type here as 
```ts
await FetchApi<RepoFile>({ query: REPO_FILE_QUERY, variables })
```
Using the `as` keyword for casting types can easily introduce bugs

Reply:
True, case of copying from other showcases